dtfs-central-api/src/services/state-machines/utilisation-report/event-handlers/mark-fee-records-as-reconciled/mark-fee-records-as-reconciled.event-handler.ts

```diff
@@ -28,29 +32,42 @@ export const handleUtilisationReportMarkFeeRecordsAsReconciledEvent = async (
   report: UtilisationReportEntity,
   { requestSource, transactionEntityManager, feeRecordsToReconcile, reconciledByUserId }: MarkFeeRecordsAsReconciledEventPayload,
 ): Promise<UtilisationReportEntity> => {
-  const feeRecordStateMachines = feeRecordsToReconcile.map((feeRecord) => FeeRecordStateMachine.forFeeRecord(feeRecord));
-  await Promise.all(
-    feeRecordStateMachines.map((stateMachine) =>
-      stateMachine.handleEvent({
-        type: 'MARK_AS_RECONCILED',
-        payload: {
-          transactionEntityManager,
-          reconciledByUserId,
-          requestSource,
-        },
-      }),
-    ),
-  );
+  try {
+    const feeRecordStateMachines = feeRecordsToReconcile.map((feeRecord) => FeeRecordStateMachine.forFeeRecord(feeRecord));
+    await Promise.all(
+      feeRecordStateMachines.map((stateMachine) =>
+        stateMachine.handleEvent({
+          type: 'MARK_AS_RECONCILED',
+          payload: {
+            transactionEntityManager,
+            reconciledByUserId,
+            requestSource,
+          },
+        }),
+      ),
+    );
 
-  const { feeRecords: allFeeRecords } = await transactionEntityManager.findOneOrFail(UtilisationReportEntity, {
-    where: { id: report.id },
-    relations: { feeRecords: true },
-  });
+    const { feeRecords: allFeeRecords } = await transactionEntityManager.findOneOrFail(UtilisationReportEntity, {
+      where: { id: report.id },
+      relations: { feeRecords: true },
+    });
 
-  if (allFeeRecords.every((record) => record.status === 'RECONCILED')) {
-    report.updateWithStatus({ status: 'RECONCILIATION_COMPLETED', requestSource });
-    return await transactionEntityManager.save(UtilisationReportEntity, report);
-  }
+    if (allFeeRecords.every((record) => record.status === 'RECONCILED')) {
+      report.updateWithStatus({ status: 'RECONCILIATION_COMPLETED', requestSource });
+      await transactionEntityManager.save(UtilisationReportEntity, report);
+      await SendReportReconciledEmail(report);
+    }
+
+    return report;
+  } catch (error) {
+    console.error(`Failed to mark fee records as reconciled - handleUtilisationReportMarkFeeRecordsAsReconciledEvent %o`, error);
 
-  return report;
+    if (error instanceof ApiError) {
+      throw TransactionFailedError.forApiError(error);
+    }
+    if (error instanceof Error) {
+      throw TransactionFailedError.forError(error);
+    }
+    throw TransactionFailedError.forUnknownError();
+  }
```

Comment:
This handling is already around this function call by nature of being within execute with SQl transaction so here we should just rethrow the original error:
```suggestion
throw error;
```